### PR TITLE
WhyLabs callback

### PR DESCRIPTION
**Pull request by jamie256**
# Add a WhyLabs callback handler

* Adds a simple WhyLabsCallbackHandler
* Add required dependencies as optional
* protect against missing modules with imports
* Add docs/ecosystem basic example

based on initial prototype from @andrewelizondo

> this integration gathers privacy preserving telemetry on text with whylogs and sends stastical profiles to WhyLabs platform to monitoring these metrics over time. For more information on what WhyLabs is see: https://whylabs.ai

After you run the notebook (if you have env variables set for the API Keys, org_id and dataset_id) you get something like this in WhyLabs:
Screenshot (443)

## Who can review?

@agola11
@hwchase17

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -89,9 +89,9 @@ gql = {version = "^3.4.1", optional = true}
 pandas = {version = "^2.0.1", optional = true}
 telethon = {version = "^1.28.5", optional = true}
 zep-python = {version="^0.25", optional=true}
+langkit = {version = "^0.0.1.dev3", optional = true}
```
Comment (hwchase17) [inline pyproject.toml:92]: this needs to be listed in the one of the extra installs (probably `all`)

**Review by hwchase17 — COMMENTED** [review]
one comment re dependencies (which we can also fix) otherwise lg

File: pyproject.toml (hunk 1 above)
Comment (jamie256) [reply pyproject.toml]: Done: Added langkit to `all` group, updated PR branch from main repo and rebased to help lower conflicts, reran `poetry lock --no-update`, reran through tests and notebook example.

File: langchain/callbacks/whylabs_callback.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+        # langkit library will import necessary whylogs libraries
+        import_langkit(sentiment=sentiment, toxicity=toxicity, themes=themes)
+
+        import whylogs as why
+        from whylogs.api.writer.whylabs import WhyLabsWriter
+        from whylogs.core.schema import DeclarativeSchema
+        from whylogs.experimental.core.metrics.udf_metric import generate_udf_schema
+
+        self.writer = WhyLabsWriter(
```
Comment (dev2049) [inline langchain/callbacks/whylabs_callback.py:57]: do we need to store this or is it only passed in to logger?

File: langchain/callbacks/whylabs_callback.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            "To use the whylabs callback manager you need to have the `langkit` python "
+            "package installed. Please install it with `pip install langkit`."
+        )
+    return langkit
+
+
+class WhyLabsCallbackHandler(BaseCallbackHandler):
+    """WhyLabs CallbackHandler."""
+
+    def __init__(
```
Comment (dev2049) [inline langchain/callbacks/whylabs_callback.py:37]: will let you decide on the best interface here, but we've found it's often nice to pass instance variable in directly to constructor (in this case just the logger) and then to have a convenience classmethod called something like `from_params` that takes in all the individual params and assembles logger and passes it in. that way you get both a convenient constructor if you want, and if somebody wants to fully control how logger is constructed and pass that in themselves, they can do that, too. also makes things a bit more robust to future changes e.g. if logger interface changes, from_params will be out-of-sync but user can still construct logger themselves and pass it in

**Review by dev2049 — COMMENTED** [review]
two optional comments, otherwise lg!

File: langchain/callbacks/whylabs_callback.py (hunk 2 above)
Comment (jamie256) [reply langchain/callbacks/whylabs_callback.py:57]: We don't need to store it with the current integration, lets simplify and not have direct access to the writer yet, removing.

File: langchain/callbacks/whylabs_callback.py (hunk 3 above)
Comment (jamie256) [reply langchain/callbacks/whylabs_callback.py:37]: Thanks, that does sound better! I will make that change within the hour, and then resolve this comment.

**Pull request merged**